### PR TITLE
Add label, better a11y to assignment menu

### LINKED ISSUE #1470: Better communication of purpose in project assignment UI
## Issue - Enhancement

As a new user, I want to know how to assign a card to a teammate. I find the current assignment UI for cards unclear in a few ways:

- When I am looking at a card, I see a name and a profile icon at the bottom. I don't know why this name and profile icon are there. Did this person _make_ the card? Are they currently _working_ on the issue? Is it on _their_ board?
- When I hover and click on a username/icon, I see a dropdown of my team members with no additional copy. I now see that the name probably isn't the creator. I am probably changing the name that will appear at the bottom of the card. But what does this have to do with the card more generally?

- **Estimated effort:** 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### PR DESCRIPTION
**Pull request by dan-f**
Fixes #1470.

Testing:
- [ ] Create a task on the team dashboard
- [ ] Open the task assignment menu
- [ ] See "Assign To:" as the menu label
- [ ] Open the inspector
- [ ] Observe that the toggle button has an appropriate `aria-label`
- [ ] Observe that the toggle button has `aria-haspopup="true"` at all times
- [ ] Observe that the toggle button has `aria-expanded="false"` when the menu is closed and `aria-expanded="true"` when the menu is open
- [ ] Observe that the menu container has `role="menu"`
- [ ] Observe that the menu container has an appropriate `aria-label`
- [ ] Observe that all menu items have `role="menuitem"`
- [ ] Observe that the `img` element in the soft invite form has an empty `alt` property, since it is decorative, and decorative images require null `alt` properties
- [ ] Observe that the `input` element in the soft invite form has an appropriate `aria-label`

### CODE REVIEW

**Comment by ackernaut** [comment]
### Test No. 1 (TA, in progress)
- [ ] Create a task on the team dashboard
- [ ] Open the task assignment menu
- [ ] See "Assign To:" as the menu label
- [ ] Open the inspector
- [ ] Observe that the toggle button has an appropriate `aria-label`
- [ ] Observe that the toggle button has `aria-haspopup="true"` at all times
- [ ] Observe that the toggle button has `aria-expanded="false"` when the menu is closed and `aria-expanded="true"` when the menu is open
- [ ] Observe that the menu container has `role="menu"`
- [ ] Observe that the menu container has an appropriate `aria-label`
- [ ] Observe that all menu items have `role="menuitem"`
- [ ] Observe that the `img` element in the soft invite form has an empty `alt` property, since it is decorative, and decorative images require null `alt` properties
- [ ] Observe that the `input` element in the soft invite form has an appropriate `aria-label`

File: src/universal/modules/outcomeCard/components/OutcomeCardAssignMenu/OutcomeCardAssignMenu.js [hunk 1]
```diff
@@ -136,12 +159,26 @@ OutcomeCardAssignMenu.propTypes = {
   assignRef: PropTypes.instanceOf(Element),
   atmosphere: PropTypes.object.isRequired,
   closePortal: PropTypes.func.isRequired,
+  styles: PropTypes.object.isRequired,
   task: PropTypes.object.isRequired,
   viewer: PropTypes.object.isRequired
 };
 
+const styleThunk = () => ({
+  label: {
```
Comment (ackernaut) [inline src/universal/modules/outcomeCard/components/OutcomeCardAssignMenu/OutcomeCardAssignMenu.js]: @dan-f did you borrow these label styles from another menu component? 👍

Comment (dan-f) [reply src/universal/modules/outcomeCard/components/OutcomeCardAssignMenu/OutcomeCardAssignMenu.js]: Yeah, I straight-up copy-pasted them from the `Menu` component. For what it's worth, this menu is already skirting around re-using the `Menu` component. Thoughts below...

Comment (ackernaut) [reply src/universal/modules/outcomeCard/components/OutcomeCardAssignMenu/OutcomeCardAssignMenu.js]: No problem. I think that if components start to get inconsistent we can scope those styles to a shared object and just `{...menuLabelStyle}` on whatever style object we need.

**Comment by dan-f** [comment]
I'll document this as a brain-dump for future reference, but I think these menus could be more flexible to work with if we could piece them together by composition rather than by configuration props.

For example, I'd like to see something along the lines of:

```js
render() {
  return (
    <Menu aria-label="Choose a Team Member to assign this task to">
      <MenuLabel>Assign To:</MenuLabel>
      <MenuItem>Anything at all can go in here!</MenuItem>
    </Menu>
  );
}
```

And this would create react elements that react would turn into DOM like this:

```html
<div role="menu" tabindex="-1" aria-label="Choose a Team Member to assign this task to">
  <div>Assign To:</div>
  <div role="menuitem">Anything at all can go in here!</div>
</div>
```

**Pull request merged**